client/src/Actions/booksAction.js

```diff
@@ -0,0 +1,374 @@
+import * as cloudKeys from '../utils/cloudinaryKeys';
+import { getUserDetails } from '../utils/getUserInfo';
+
+
+// *********************************************************** //
+// DEFINE ACTION CREATOR AND MAKE API CALL FOR GET ALL BOOKS** //
+// *********************************************************** //
+
+export function getAllBooks(books) {
+  return {
+    type: types.GET_ALL_BOOKS,
+    books
+  }
+}
+
+// make a request for books
+export function loadAllbooks() {
+  return (dispatch) => {
+    return axios.get(apiRoutes.books).then( response => {
+      dispatch(getAllBooks(response.data.books));
+    }).catch(error => {
+      console.log(error)
+    })
+  }
+}
+
+export function getBookId(bookid) {
+  return {
+    type: types.GET_BOOK_ID,
+    bookid
+
+  }
+}
+
+// ********************************************************** //
+// *DEFINE ACTION CREATOR AND MAKE API CALL FOR BORROW BOOKS* //
+// ********************************************************** //
+
+export function borrowBookAction(bookDetails) {
+  return {
+    type: types.BORROW_BOOK,
+    bookDetails
+  }
+}
+
+
+export function borrowBook(bookDetails) {
+  return dispatch => {
+    return axios.post(`${apiRoutes.userProfile}/${getUserDetails().userId}/books`, 
+      bookDetails, {
+        headers: {'Authorization': getUserDetails().savedToken}
+      })
+      .then(response => {
+        dispatch(borrowBookAction(response.data))
+      })
+      .catch(error => {
+        throw (error)
+      })
+  }
+}
+
+
+// ********************************************************** //
+// *DEFINE ACTION CREATOR & MAKE API CALL FOR GET USER BOOKS* //
+// ********************************************************** //
+
+/***
+ * 
+ * @param { object } fetchedBooks
+ * @returns { object } Action Type and fetched books 
+ */
+export function userBooks(fetchedBooks) {
+  return {
+    type: types.FETCTH_USER_BOOKS,
+    fetchedBooks
+  }
+}
+
+export function getUserBooks() {
+  return dispatch => {
+    return axios.get(`${apiRoutes.userProfile}/${getUserDetails().userId}/books`,  
+      {
+        headers: {'Authorization': getUserDetails().savedToken}
+      }
+      ).then(response => {
+        dispatch(userBooks(response.data));
+      })
+      .catch(error => {
+        throw (error)
+      })
+  }
+}
+
+
+// ********************************************************** //
+// *DEFINE ACTION CREATOR AND MAKE API CALL FOR RETURN BOOKS* //
+// ********************************************************** //
+
+export function returnBookAction(bookid) {
+  return{
+    type: types.RETURN_BOOK,
+    bookid
+  }
+}
+
+export function returnBook(bookid) {
+  return dispatch => {
+    return axios.put(`${apiRoutes.userProfile}/${getUserDetails().userId}/books`, 
+    bookid,
+    {headers: {'Authorization': getUserDetails().savedToken}})
+    .then(response => {
+      dispatch(returnBookAction(response.data.message));
+    })
+    .catch(error => {
```

Comment:
So what happens if there's an error

Reply:
If there is an error, it will be displayed in the component where i am calling the function.